File: Hotel-backend/src/main/java/com/acuario/hotel/service/ServicioService.java
package com.acuario.hotel.service;

import com.acuario.hotel.dto.ServicioRequestDTO;
import com.acuario.hotel.dto.ServicioResponseDTO;
import java.util.List;

public interface ServicioService {
	
	List<ServicioResponseDTO> listar();
	
	ServicioResponseDTO crear(ServicioRequestDTO dto);
	
	ServicioResponseDTO obtener(Long id);
	
	ServicioResponseDTO actualizar(Long id, ServicioRequestDTO dto);
	
	void eliminar(Long id);
}

File: hotel-acuario-front/src/app/services/servicio.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Servicio } from '../models/servicio.model';
import { environment } from '../../environments/environment.development';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root',
})
export class ServicioService {
  private readonly base = `${environment.apiBase}/servicios`;

  constructor(private http: HttpClient) {}

  listar(): Observable<Servicio[]> {
    return this.http.get<Servicio[]>(this.base);
  }

  obtener(id: number): Observable<Servicio> {
    return this.http.get<Servicio>(`${this.base}/${id}`);
  }

  crear(servicio: Servicio): Observable<Servicio> {
    return this.http.post<Servicio>(this.base, servicio);
  }

  actualizar(id: number, servicio: Servicio): Observable<Servicio> {
    return this.http.put<Servicio>(`${this.base}/${id}`, servicio);
  }

  eliminar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.base}/${id}`);
  }
}




